### PR TITLE
Move to Dynamoid for data persistence

### PR DESCRIPTION
**Pull request by theseanything**
This disables active record and adds dynamoid to persist the FromResponse in DynamoDB.  We are switching from Postgres to using DynamoDB (a AWS managed NoSQL DB) to store the data. Doing this because we can set IAM permissions so the app is allowed to store data, but cannot read it (for security).

### CODE REVIEW

**Comment by tlwr** [comment]
Can remove the `pg` gem and the service in `.travis.yml`

Will probably need to add dynamodb localstack in travis for tests, or use a stub

**Comment by tlwr** [comment]
Most recent changes look 🥇

**Review by bilbof — COMMENTED** [review]
Please could you update the readme steps for running locally?

File: app/controllers/coronavirus_form/check_answers_controller.rb [hunk 1]
```diff
@@ -15,8 +15,13 @@ def show
   def submit
     submission_reference = reference_number
 
-    session[:reference_number] = submission_reference
-    FormResponse.create(form_response: session)
+    session[:reference_id] = submission_reference
```
Comment (bilbof) [inline app/controllers/coronavirus_form/check_answers_controller.rb:19]: I wonder if we could set this right at the beginning of the journey, then we could use it for logging / Sentry error context. Of course not for this PR :)

File: spec/support/dynamoid_helper.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module DynamoidReset
+  def self.all
+    Dynamoid.adapter.list_tables.each do |table|
+      # Only delete tables in our namespace
```
Comment (bilbof) [inline spec/support/dynamoid_helper.rb:4]: Is it possible for this constraint to be enforced outside of the application code?

Comment (theseanything) [reply spec/support/dynamoid_helper.rb:4]: Which constrain sorry? This shouldn't be too much of an issue as it just a database cleaner for when we run the tests.

File: features/support/env.rb [hunk 3]
```diff
@@ -32,11 +32,11 @@
 
 # Remove/comment out the lines below if your app doesn't have a database.
 # For some databases (like MongoDB and CouchDB) you may need to use :truncation instead.
-begin
-  DatabaseCleaner.strategy = :transaction
-rescue NameError
-  raise "You need to add database_cleaner to your Gemfile (in the :test group) if you wish to use it."
-end
+# begin
```
Comment (huwd) [inline features/support/env.rb:35]: Can we just remove this stuff?

File: config/environments/production.rb [hunk 4]
```diff
@@ -83,7 +83,7 @@
   end
 
   # Do not dump schema after migrations.
-  config.active_record.dump_schema_after_migration = false
+  # config.active_record.dump_schema_after_migration = false
```
Comment (huwd) [inline config/environments/production.rb:86]: can we remove this?

File: config/environments/development.rb [hunk 5]
```diff
@@ -35,10 +35,10 @@
   config.active_support.deprecation = :log
 
   # Raise an error on page load if there are pending migrations.
-  config.active_record.migration_error = :page_load
+  # config.active_record.migration_error = :page_load
 
   # Highlight code that triggered database queries in logs.
-  config.active_record.verbose_query_logs = true
+  # config.active_record.verbose_query_logs = true
```
Comment (huwd) [inline config/environments/development.rb:41]: can we remove this?

File: config/environments/development.rb [hunk 6]
```diff
@@ -35,10 +35,10 @@
   config.active_support.deprecation = :log
 
   # Raise an error on page load if there are pending migrations.
-  config.active_record.migration_error = :page_load
+  # config.active_record.migration_error = :page_load
```
Comment (huwd) [inline config/environments/development.rb:38]: can we remove this?

File: config/application.rb [hunk 7]
```diff
@@ -6,7 +6,7 @@
 # Pick the frameworks you want:
 require "active_model/railtie"
 require "active_job/railtie"
-require "active_record/railtie"
+# require "active_record/railtie"
```
Comment (huwd) [inline config/application.rb:9]: can we remove this?

File: features/support/env.rb [hunk 8]
```diff
@@ -56,4 +56,4 @@
 # Possible values are :truncation and :transaction
 # The :transaction strategy is faster, but might give you threading problems.
 # See https://github.com/cucumber/cucumber-rails/blob/master/features/choose_javascript_database_strategy.feature
-Cucumber::Rails::Database.javascript_strategy = :truncation
+# Cucumber::Rails::Database.javascript_strategy = :truncation
```
Comment (huwd) [inline features/support/env.rb:59]: can we remove this?

**Comment by theseanything** [comment]
@huwd I'm leaving in those commented lines just so people understand that ActiveRecord isn't being used.

**Comment by huwd** [comment]
If you update the readme, i'll pull this branch and give it a test locally?

**Review by huwd — APPROVED** [review]

**Pull request merged**